## Домашнее задание к лекции «Случайные события. Случайные величины»

> Текст задания см. на скриншоте http://prntscr.com/uh35yg

Смоделировать игру против лотерейного автомата типа “777”. Игрок платит 1 руб., после чего выпадает случайное целое число, равномерно распределенное от 0 до 999. При некоторых значениях числа игрок получает выигрыш.

* 777: 200 руб.
* 999: 100 руб.
* 555: 50 руб.
* 333: 15 руб.
* 111: 10 руб.
* *77: 5 руб.
* **7: 3 руб.
* *00: 2 руб.
* **0: 1 руб.

Вопросы:

    Выгодна ли игра игроку?
    Сколько в среднем приобретает или теряет игрок за одну игру?

**Дополнительное задание повышенной сложности. **

Теоретически рассчитать средний выигрыш (проигрыш) и сравнить с результатами моделирования

    *– любая цифра



In [1]:
import pandas as pd
import numpy as np


In [2]:
%time
wallet = 0 #первый рубль не считаем,как и последующие в случае . 
plays = 10000 #количество игр
win_lose_ratio = pd.DataFrame({"results":[]}) #собираем статистику по выигрыш/проигрыш, в бинарном формате
bets = {777:200,999:100,
        555:50,333:15,
        111:10
       } #Выигрыш крупных ставок

for i in range(plays): 
    bet = np.random.randint(0,1000)
    wallet -= 1
    if bet in bets.keys():
        wallet += bets[bet]
        win_lose_ratio = win_lose_ratio.append({"results":"Win"},ignore_index=True)
    elif (bet - 77) % 100 == 0: # *77: 5 руб.
        wallet += 5
        win_lose_ratio = win_lose_ratio.append({"results":"Win"},ignore_index=True)    
    elif (bet-7) % 10 == 0:  # **7: 3 руб.
        wallet += 3
        win_lose_ratio = win_lose_ratio.append({"results":"Win"},ignore_index=True)
    elif bet % 100 == 0: #*00: 2 руб.
        wallet += 2
        win_lose_ratio = win_lose_ratio.append({"results":"Win"},ignore_index=True)
    elif bet % 10 == 0: #**0: 1 руб
        wallet += 1
        win_lose_ratio = win_lose_ratio.append({"results":"Win"},ignore_index=True)
    else:
        win_lose_ratio = win_lose_ratio.append({"results":"Lost"},ignore_index=True)


CPU times: user 2 µs, sys: 1 µs, total: 3 µs
Wall time: 4.05 µs


In [3]:
total = win_lose_ratio['results'].value_counts()
win_sums = total["Win"] / plays * 100
print(f"на {plays} игр пришлось лишь {win_sums}% выигрыша. Общая сумма\
 в кошельке игравшего составляет {wallet} рублей.\
 В данном наборе испытаний на 1 игру игрок получает {wallet/plays}\
 рублей выигрыша/убытка")

на 10000 игр пришлось лишь 20.46% выигрыша. Общая сумма в кошельке игравшего составляет -3061 рублей. В данном наборе испытаний на 1 игру игрок получает -0.3061 рублей выигрыша/убытка


#### Выгодна ли игра игроку?
нет, не выгодна. На 10000 игр только ~20% (20,46%) оказались выигрышными, а общий результат в любом случае был отрицательным. В конкретном примере равный -3061. 
#### Сколько в среднем приобретает или теряет игрок за одну игру?
-0,3061 рублей. 